src/Umbraco.Web.Common/DependencyInjection/UmbracoBuilderExtensions.cs

```diff
@@ -329,9 +329,35 @@ public static IUmbracoBuilder AddHelpers(this IUmbracoBuilder builder)
     [Obsolete("This is not necessary any more. This will be removed in v17")]
     public static IUmbracoBuilder AddWebServer(this IUmbracoBuilder builder)
     {
+        builder.Services.Configure<KestrelServerOptions>(options =>
+        {
+            options.AllowSynchronousIO = true;
+        });
+
+        try
+        {
+            // See https://github.com/umbraco/Umbraco-CMS/pull/17886. This is a workaround for non-windows machines
+            // they won't have IIS available and trying to set this option will throw an exception.
+            //
+            // We're deferring this call to a method because if we just try to set the options here, we still get a
+            // TypeLoadException on non-windows machines.
+            // This workaround came from this comment: https://stackoverflow.com/a/3346975
+            AllowSynchronousIOForIIS(builder);
+        }
+        catch (Exception)
+        {
+            // Ignoring this exception because it's expected on non-windows machines
+        }
         return builder;
     }
 
+    private static void AllowSynchronousIOForIIS(IUmbracoBuilder builder) =>
```

Comment:
Maybe you need a `[MethodImpl(MethodImplAttributes.NoInlining)]` here?  At least that's what's suggested in the workaround we found and you linked above.  Otherwise I assume there's a risk of a build or runtime optimization being applied that would trigger the issue again,

Reply:
It was enough to just move this to a method. I built everything with: `dotnet build && dotnet pack --configuration Release --output .\release\` which is basically what we do when releasing the CMS. I tested before (v15.1.0), not moving this part to a method and moving this part to a method. The last attempt succeeded.

In any case, adding the attribute works just as well, can't hurt. So I've added that change too!